src/ansys/dpf/post/faces.py

```diff
@@ -0,0 +1,158 @@
+        return self._face.id
+
+    @property
+    def index(self) -> int:
+        """See :py:meth:`ansys.dpf.core.faces.Face.index`."""
+        return self._face.index
+
+    @property
+    def nodes(self) -> NodeListByIndex:
+        """See :py:meth:`ansys.dpf.post.nodes.NodeListByIndex`."""
+        return NodeListByIndex(self._face.nodes)
+
+    @property
+    def num_nodes(self) -> int:
+        """See :py:meth:`ansys.dpf.core.faces.Face.n_nodes`."""
+        return self._face.n_nodes
+
+    @property
+    def type_info(self) -> ElementType:
+        """Gets an element descriptor, See :py:meth:`ansys.dpf.core.faces.Face.id`."""
+        return ElementType(self._face.type.value)
+
+    @property
+    def type(self) -> core_elements.element_types:
+        """Returns the Element Type."""
+        return self._face.type
+
+    @property
+    def to_node_connectivity(self) -> List[int]:
+        """See :py:meth:`ansys.dpf.core.faces.Face.connectivity`."""
+        return self._face.connectivity
+
+    def __repr__(self) -> str:
+        """Returns string representation of a Face."""
+        return f"Face(type={self.type},index={self.index},id={self.id})"
+
+    def __str__(self) -> str:
+        """Returns string representation of a Face."""
+        return str(self._face)
+
+
+class _FaceList(ABC):
+    """Iterator class for the FaceList."""
+
+    def __init__(self, face_list: core_faces.Faces):
+        """Constructs an Iterator from a face list."""
+        self._face_list = face_list
+        self._idx = 0
+
+    def __next__(self) -> Face:
+        """Returns the next Face in the list."""
+        if self._idx >= len(self._face_list):
+            raise StopIteration
+        ret = self[self._idx]
+        self._idx += 1
+        return ret
+
+    def __getitem__(self, index: int) -> Face:
+        """Returns a post.Face based on an index in the current list."""
+        return Face(self._face_list[index])
+
+    def __len__(self) -> int:
+        """Returns the number of faces in the list."""
+        return self._face_list.n_faces
+
+    def __repr__(self) -> str:
+        """Returns a string representation of _FaceList object."""
+        return f"{self.__class__.__name__}({self}, __len__={len(self)})"
+
+    def _short_list(self) -> str:
+        _str = "["
+        if self.__len__() > 3:
+            _fst = Face(self._face_list[0]).type_info.name
+            _lst = Face(self._face_list[len(self) - 1]).type_info.name
+            _str += f"{_fst}, ..., {_lst}"
+        else:
+            face_list = [Face(self._face_list[idx]) for idx in range(len(self))]
+            _str += ", ".join(map(lambda el: el.type_info.name, face_list))
+        _str += "]"
+        return _str
+
+    def __str__(self) -> str:
+        """Returns a string representation of a _FaceList object."""
+        return self._short_list()
+
+    @abstractmethod
+    def __iter__(self):  # pragma: no cover
+        """Returns the object to iterate on."""
```

Comment:
To add something to the function.

```suggestion
    def __iter__(self):  # pragma: no cover
        """Returns the object to iterate on."""
        raise NotImplemented
```